### ISSUE TITLE
Docker image for Raspberry Pi

### ISSUE THREAD

**Issue by koenvervloesem** [juanluisbaptiste/docker-postfix#23]
I noticed that there's only an amd64 image of this container on Docker Hub.

I just checked out the `migrate_to_alpine` branch, built the container with `docker build -t postfix .` and I can confirm that the resulting container just works on a Raspberry Pi running Raspbian 10.3. Could you also publish arm builds of the image to Docker Hub?

**Comment by juanluisbaptiste** [comment]
Hi,

Is not on my plans right now to support other architectures, but help is welcom, you can send a PR to add support for other architectures.

**Comment by koenvervloesem** [comment]
This actually doesn't need any PR, but some changes in the image building and uploading.

Here's how I tried to do this:

First some preparation on the build host (which is a amd64 machine), as explained in a [blog post from Docker](https://www.docker.com/blog/multi-platform-docker-builds/):

```
docker run --privileged --rm docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
sudo systemctl restart docker.service
mkdir ~/.docker/cli-plugins
wget https://github.com/docker/buildx/releases/download/v0.4.1/buildx-v0.4.1.linux-arm64 -O ~/.docker/cli-plugins/docker-buildx
chmod a+x ~/.docker/cli-plugins/docker-buildx
```
I only created images for the `migrate_to_alpine` branch. Build the images for different platforms and push them:

```
docker buildx build --platform linux/amd64 -t koenvervloesem/postfix:amd64 .
docker push koenvervloesem/postfix:amd64
docker buildx build --platform linux/arm/v7 -t koenvervloesem/postfix:armv7 .
docker push koenvervloesem/postfix:armv7
docker buildx build --platform linux/arm64/v8 -t koenvervloesem/postfix:armv8 .
docker push koenvervloesem/postfix:armv8
```

Then create and push a manifest so the alpine tag refers to the right tag for your architecture:

```
docker manifest create koenvervloesem/postfix:alpine koenvervloesem/postfix:amd64 koenvervloesem/postfix:armv7 koenvervloesem/postfix:armv8
docker manifest push koenvervloesem/postfix:alpine
```

You can try the result, a multi-arch Docker build for postfix, from [koenvervloesem/postfix:alpine](https://hub.docker.com/r/koenvervloesem/postfix).

However, here's where I'm stuck. Exactly the same process worked for [another project](https://github.com/jpmens/mqttwarn/issues/424). But when I do a `docker pull koenvervloesem/postfix:alpine` on Raspbian and run this container, I get the `standard_init_linux.go:211: exec user process caused "exec format error"` message. On my Ubuntu amd64 machine, it works.

It would be awesome if this multi-arch build process could be made to work and integrated in this project's CI so Raspberry Pi users can easily use your image.

Can anyone else try this process and see whether it works there? Maybe I have made a mistake in one of these steps.

**Comment by revsystem** [comment]
I succeeded to create a multi-arch Docker image for postfix.
I created this image on MacBook Pro. And I'm using this image on Raspberry Pi 3B+. 

DockerHUB
https://hub.docker.com/r/revsystem/postfix/tags?page=1&ordering=last_updated

My repository is here.
https://github.com/revsystem/docker-postfix

@koenvervloesem @juanluisbaptiste

**Comment by juanluisbaptiste** [comment]
Hi @revsystem thank you, could you point me to the changes done to add this support ?

TY

**Comment by revsystem** [comment]
@juanluisbaptiste 
I didn't change anything.
I just only executed this command.

```
$ docker login
$ docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 -t revsystem/postfix:latest --push .
```

**Comment by juanluisbaptiste** [comment]
I merged PR #52  which will create images for other architectures different to x86_64, currently building.
